### PR TITLE
clippy suggestions and warning removal

### PR DESCRIPTION
**Pull request by clux**
fixes:

1. unused imports for docs (made explicit, or removed when unused afaikt)
2. unused imports +  attr style
3. `clone_from` rather than assigning clone
4. chrono's duration's now needs unwrap which is not const - https://github.com/chronotope/chrono/issues/1491

To see what the warnings looked like an hour ago. Look at the files annotated in the previous PR: https://github.com/kube-rs/kube/pull/1435/files#file-kube-client-src-client-mod-rs-L10

### CODE REVIEW

File: kube-client/src/client/auth/mod.rs [hunk 1]
```diff
@@ -621,7 +633,7 @@ mod test {
     #[tokio::test]
     #[ignore = "fails on windows mysteriously"]
     async fn exec_auth_command() -> Result<(), Error> {
-        let expiry = (Utc::now() + Duration::seconds(60 * 60)).to_rfc3339();
+        let expiry = (Utc::now() + SIXTY_SEC).to_rfc3339();
```
Comment (clux) [inline kube-client/src/client/auth/mod.rs:636]: (this time was arbitrary and the test is broken anyway)

File: kube-runtime/src/reflector/object_ref.rs [hunk 2]
```diff
@@ -1,8 +1,10 @@
 use derivative::Derivative;
 use k8s_openapi::{api::core::v1::ObjectReference, apimachinery::pkg::apis::meta::v1::OwnerReference};
+#[allow(unused_imports)] // docs
```
Comment (nightkr) [inline kube-runtime/src/reflector/object_ref.rs:3]:
```suggestion
#[cfg(doc)]
```

File: kube/src/mock_tests.rs [hunk 3]
```diff
@@ -61,6 +61,7 @@ async fn timeout_after_1s(handle: tokio::task::JoinHandle<()>) {
 /// Scenarios we test for in ApiServerVerifier above
 enum Scenario {
     PaginatedList,
+    #[allow(dead_code)]
     RadioSilence,
```
Comment (nightkr) [inline kube/src/mock_tests.rs:65]: Does nothing use this variant then?

Comment (clux) [reply kube/src/mock_tests.rs:65]: it's matched on in a select, but we don't instantiate it atm, was planning on maybe using it in a test, as it's a common mock base scenario and it felt weird to not have it

Comment (nightkr) [reply kube/src/mock_tests.rs:65]: Fair, I'd maybe mention that in a comment.

Comment (clux) [reply kube/src/mock_tests.rs:65]: minor comment added in https://github.com/kube-rs/kube/pull/1436/commits/f49d4edb77e6c71f8f6fa4eebe644c547198bef4

**Review by nightkr — APPROVED** [review]

**Pull request merged**